### ISSUE TITLE
Are results expected?

### ISSUE THREAD

**Issue by foreversolar** [ziansu/EBSNN-Reproduce#1]
First, this is an excellent reproduction work with better coding quality for reading and extension. 

I found 2 issues that (may exist as original work or this reproduced work)
- data preprocess: according to preprocessing.ipynb and data.py, it shows that the packet is not segmented according to the protocol (as TCP and IP should be first divided and then segmented and padded). It is also not masked for IP address and port number, which leads to the model overfitting. 
- model testing: I found that the evaluation work is done on the val dataset but not test dataset

Looking forward to your response and understanding.

**Comment by ziansu** [comment]
This is just a course work project, so there might be some unrealistic settings.
- About the preprocessing, I think I generally followed the original work.
- About the evaluation, you can use validation set for model selection and get the results on test set. If the distributions are similar, I think they will give similar results.

**Comment by foreversolar** [comment]
Thanks for your reply and your great work in reproducing. It's really good. 

- preprocessing: I think the author just preprocessed the data and loaded them to train and test. So the code is not public on the original repo. So, this preprocessing work is not consistent with the paper (just to note it)
- results: I try to employ the final checkpoint during the train to test and it works.

**Comment by ziansu** [comment]
Thank you for pointing out the problem with the preprocessing. Good luck with your project!
